### PR TITLE
Use new IPC compliant generators with db-parts queries.

### PR DESCRIPTION
**Pull request by m-hilgendorf**
Fixes bug where courtyards had the wrong orientation (and many other potential bugs)

### CODE REVIEW

File: utils/land-patterns.stanza [hunk 1]
```diff
@@ -1676,6 +1676,46 @@ public defn two-pin-chip-landpattern (length:Toleranced,
...
 
+public defn parse-chip-package-code (code:String) -> Dims : 
+  ; helper function to insert decimal point after first character.
+  defn insert-decimal (s:String) -> String :
+    to-string $ "%_.%_" % [s[0], s[1 to false]] 
+
+  ; validate the code :
+  ;   - must be all numeric (TBD: extract metric/imperial information before calling)
+  ;   - must be 4, 5, or 6 characters long
+  if find({not digit?(_)}, code) is-not False or
```
Comment (ariel-hi) [inline utils/land-patterns.stanza:1687]: The first condition is a double negative, so that makes it hard to follow. The documentation above is great!
`not all?(digit?, code)`

File: utils/land-patterns.stanza [hunk 2]
```diff
@@ -1676,6 +1676,46 @@ public defn two-pin-chip-landpattern (length:Toleranced,
...
+  name = to-string $ "%_%_%_" % [
+    code, "i" when imperial? else "", "p" when polarized? else ""]
+  description = to-string $ "Auto generated %_ (%_)%_ land pattern." % [
+    code, "Imperial" when imperial? else "Metric", ", Polarized" when polarized? else ""]
+  var dims = parse-chip-package-code(code)
+  if imperial? :
+    dims = Dims(x(dims) * 25.4, y(dims) * 25.4)
+  val package-length = tol(x(dims), 0.0)
+  val package-width  = tol(y(dims), 0.0)
+  val lead-length    = tol(x(dims) * 0.33, 0.0) ; total guess
```
[the remark this answers is not in the document]
Comment (electricfungi) [reply utils/land-patterns.stanza:1713]: Add a switch function based on package, and support for packages in the database: ["0201" "02016" "0202" "0302" "0303" "0402" "0404" "0503" "0505" "0603" "0612" "0805" "1206" "1210" "1218" "1812" "2010" "2512" "2525" "2615" "2616" "3920" "4122" "4823" "5329" "6030"]
Here is a link for some teminations: https://sh.kemet.com/Lists/TechnicalArticles/Attachments/29/f2100e.pdf
other reference termination information can come from datasheets of passive components.
Avoid adding package sizes not on the list - especially smaller, they need to be specific. 
Add a TODO: connect with database for actual terminal connections
